app/components/common/error/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+angular.module('odca.error', [])
+  .config(['$httpProvider', function ($httpProvider) {
+    $httpProvider.interceptors.push('errorHttpInterceptor');
+  }])
+  .run(['$window', 'error', function ($window, error) {
+    $window.onerror = onError;
+
+    function onError (message, url, lineNo, columnNo, err) {
+      error(err, message, true);
+    }
+  }])
+  .directive('odcaErrorMessage', function () {
+    return {
+      restrict: 'E',
+      controller: ErrorMessageController,
+      controllerAs: '$ctrl',
+      bindToController: true,
+      template: require('./error_message.html')
+    };
+  })
+  .decorator('$exceptionHandler', ['error', function (error) {
+    return exceptionHandler;
+
+    function exceptionHandler (err, cause) {
+      error(err, cause);
+    }
+  }])
+  .factory('error', ['$injector', '$log', function ($injector, $log) {
+    return error;
+
+    function error (err, cause, fatal) {
+      var $rootScope = $injector.get('$rootScope'); // avoids a circular dependency with the $exceptionHandler
+      var $location = $injector.get('$location'); // avoids a circular dependency with the $rootScope
+      var messages = [err];
+      if (cause) {
+        messages.unshift(cause);
+      }
+
+      $log.error.apply(null, messages);
+      $rootScope.$emit('odca.error-message', cause);
+
+      if ($location.host().indexOf('localhost') === -1) {
+        ga('send', 'exception', {
+          exDescription: cause || err.message || err,
+          exFatal: !!fatal,
+        });
+      }
+    }
+  }])
+  .factory('errorHttpInterceptor', ['error', function (error) {
+    return {
+      responseError: function errorHttpInterceptor (response) {
+        //TODO maybe check response.status and redirect to 404?
+        error(new Error(response.statusText));
```

Comment:
Ah, good point. In that case, I'm thinking we don't need any handler here at all, that way the default http handler will just throw the error and the global handler will catch it.